anxcloud/resource_dns_record.go

```diff
@@ -0,0 +1,223 @@
+package anxcloud
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"go.anx.io/go-anxcloud/pkg/api"
+	"go.anx.io/go-anxcloud/pkg/api/types"
+	clouddnsv1 "go.anx.io/go-anxcloud/pkg/apis/clouddns/v1"
+	"go.anx.io/go-anxcloud/pkg/utils/object/compare"
+)
+
+var syncDNSRecordOps sync.Mutex
+var dnsRecordSkipMutexLock = providerContextKey("dns-record-skip-mutex-lock")
+
+func resourceDNSRecord() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceDNSRecordCreate,
+		ReadContext:   resourceDNSRecordRead,
+		UpdateContext: resourceDNSRecordUpdate,
+		DeleteContext: resourceDNSRecordDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Timeouts: &schema.ResourceTimeout{
```

Comment:
that's quite long timeouts, especially for Create and Update - do we need them to be that long?